### ISSUE TITLE
MySQL: Migration from 2.0.0-beta1 to 2.0.0 fails

### ISSUE THREAD

**Issue by derboblan** [adobo/agendav#197]
Executing dry run of migration up to 20160407194955 from 20140812203419

  ++ migrating 20150804202842

Creating table principals
Migration 20150804202842 failed during Execution. Error Unknown database type enum requested, Doctrine\DBAL\Platforms\MySqlPlatform may not support it.

                                                                                                   
  [Doctrine\DBAL\DBALException]                                                                    
  Unknown database type enum requested, Doctrine\DBAL\Platforms\MySqlPlatform may not support it.

**Comment by adobo** [comment]
Do you have tables from any other application on the same database? AgenDAV does not use any `ENUM` columns, and after reading [a blog post about the error message](https://wildlyinaccurate.com/doctrine-2-resolving-unknown-database-type-enum-requested/) looks like Doctrine has found more tables on the same database that actually use `ENUM`.

Is that the case?

**Comment by derboblan** [comment]
Yeah that is true, i saw the post aswell...but still there is not even the slightest hint, which file needs to be patched to work properly. Omg am I the first person trying to update this on a shared database?

**Comment by adobo** [comment]
@derboblan can you try the patch at https://gist.github.com/adobo/9a60393104e803b2e6ff4bba3a6b4fbd ? It uses the approach suggested by the blog post above.

**Comment by derboblan** [comment]
No, this didn't help. I solved it by adding `$this->_platform->registerDoctrineTypeMapping('enum', 'string');` to the end of the constructor in web/vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php. Then there iwas another problem with an already existing table, that i just deleted before migrating.

Not sure, if I would consider the bug closed though.

**Comment by pixelbrackets** [comment]
As far as i can tell this was an issue in doctrine 2.x. We now use 3.0 where this is handled differently. Shoudl be gone in AgendDAV 3.0
